plasma_framework/contracts/mocks/exits/dummyVaults/SpyEthVaultForExitGame.sol

```diff
@@ -0,0 +1,18 @@
+pragma solidity ^0.5.0;
+
+import "../../../src/vaults/EthVault.sol";
+import "../../../src/framework/PlasmaFramework.sol";
+
+contract SpyEthVaultForExitGame is EthVault {
+    event DummyEthWithdraw(
```

Comment:
1. 
> Don't we want such an event in production code as well?

My intuition is that this event would be useful.

Following is my current reasoning on commit message, but if you have anything to add or have better idea please let me know. I would like to reason about why we add this so if we find it reluctant and nobody is using it we can check our motivation and remove that.

```
This commit adds 'EthWithdrawn' and 'Erc20Withdrawn' event when
the 'withdraw' function is called in vaults. We might be interested
in monitoring withdraw events or rely on this to understand an user's
interaction with Plasma.
```

2. 
> If we added a similar event to the non-mock, we then would probably not need the mock anymore.

yes and no. The tradeoff is we want to maintain test with more complex setup or we want to maintain the extra mock code. We would need to prepare fund for vault and bypass all checks on the function. (It checks that must be called from an exit game and the exit game must not be quarantined.) Anyway, in the case here, I tried. We can do it for ERC20 but for ETHVault, we would need a way to send ETH to the vault. I am not sure if there is way to do that aside from creating a test function that is `payable`, which would lead to the need of mock again? Given that I revert back my change and keep the mock object here.